parse.py

```diff
@@ -0,0 +1,59 @@
+res: str = ""
+
+class Line:
+    def __init__(self, line) -> None:
+        """
+            receives a line of readme and updates the link to leetcode
+        """
+        self.__lst = [i.strip() for i in line.split('|') if i]
+        self.__title = get_title(line)
+        self.__root = "https://leetcode.com/problems/"
+        self.__route = self.__lst[1].replace(" ", "-").lower()
```

Comment:
Try something like this in order to fix the broken links from special characters (e.g. 308 and 381)

```suggestion
        self.__route = self.__lst[1].replace(" ", "-").lower()
        import re
        
        # Remove duplicate hyphens and special characters
        rep = lambda m : ("", "-") ["-" in m.group(0)]
        self.__route = re.sub(r"\W+", rep, self.__route)
```